### PR TITLE
Upgrade Helix dependency to 1.4.0

### PR DESCRIPTION
**Pull request by abhioncbr**
**Labels**
- `dependencies`

**Description**
- As per the [issue](https://github.com/apache/pinot/issues/13472), upgrading the helix dependency to `1.4.0`

### CODE REVIEW

**Comment by abhioncbr** [comment]
> @abhioncbr - Do we already know of any significant changes since the last upgrade ?

I have to look more into details, in the first look seems like nothing much signifacnt changed in Helix. We are getting failures in the restart kinds of use cases. I will dog more. Thanks

**Comment by jackjlli** [comment]
Thanks @abhioncbr for working on it! Please keep us posted about the testing results against the new Helix release.

**Comment by abhioncbr** [comment]
We are seeing a lot of test failures in Pinot because of the changes in this [PR](https://github.com/apache/helix/pull/2782/files#diff-321cf13f9b2ecee61b329273b373b7cac7ec1b2e4b99b66c3d045847f2c1f040L94) in Helix

I am raising the issue of Helix removing the condition or working around this.
cc: @jackjlli / @Jackie-Jiang

**Comment by Jackie-Jiang** [comment]
@siddharthteotia @jackjlli Any update on the internally testing? Is there new introduced behavior change in Helix? A lot of tests are failing

**Comment by jackjlli** [comment]
Hi @abhioncbr, after looking at the exception message in the test a bit, I think in our tests when a table gets newly created, we should add some validation checks to make sure if the table is fully ready before testing out anything else.

For example, in `testListConfigs()` in the `TableConfigsRestletResourceTest` class, we should add the following validation logic right after the table creation request is sent (in your PR, it should be added after Line 362):
```
    String getExternalViewUrl = DEFAULT_INSTANCE.getControllerRequestURLBuilder().forTableExternalView(tableName1);

    TestUtils.waitForCondition(aVoid -> {
      // should wait for both realtime and offline table external view to be live.
      try {
        String externalViewJsonResponse = sendGetRequest(getExternalViewUrl);
        TableViews.TableView externalView = JsonUtils.stringToObject(externalViewJsonResponse, TableViews.TableView.class);
        return externalView._offline != null;
      } catch (IOException e) {
        return false;
      }
    }, 30_000L, "Failed to find the table creation from the ExternalView");
```
Basically, we should never expect that all the znodes are created once the table creation request was made, as any updates on external view should be done by Helix controller.
Same for other places where the table creation request was made in the same test class. @abhioncbr can you address that in your PR?

**Comment by Jackie-Jiang** [comment]
@jackjlli But this doesn't explain why the tests are not failing before the upgrade. For `testListConfigs()`, why do we need to wait for EV to appear? Table list API should only check the IS instead of EV, unless there are behavior changes in Helix

**Comment by abhioncbr** [comment]
> @abhioncbr , shall we consider upgrading Pinot to https://github.com/apache/helix/releases/tag/helix-1.4.3 version?

Sure, we can definitely try that.

**Comment by abhioncbr** [comment]
closing in favour of https://github.com/apache/pinot/pull/15063
